gi_core/include/common.hpp

```diff
@@ -14,17 +19,44 @@ namespace yagit::core
 {
 	using size_t = std::size_t;
 	using std::error_code;
+	using std::unique_ptr;
 	using std::shared_ptr;
 	using std::enable_if_t;
 	using std::optional;
 	using std::nullopt;
+	using std::array;
 	using std::true_type;
 	using std::false_type;
 	using std::declval;
+	using std::index_sequence;
+	using std::make_index_sequence;
+	using std::variant;
+	using std::numeric_limits;
 #ifndef ALLOCATE_AT_LEAST_NOT_DEFINED
 	using std::allocation_result;
 #endif
 
+	template<typename T>
+	using view = T*;
+
+	template<typename T>
+	using const_view = const T*;
+
+	template<size_t N>
+	constexpr bool is_power_of_two_v = N && !(N & (N - 1));
```

Comment:
Explain from where did you get that code to check if something is power of 2 ?
If from stack overflow, then paste here the link to the proper page

Reply:
Simple binary trick learned some time ago, its UB-free as it is evaluated inside constexpr expression which doesn't allow any undefined behaviour by C++ standard

Reply:
Add as a comment